catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/AccessControlPreQueryPluginTest.java

```diff
@@ -0,0 +1,320 @@
+        is(
+            equalTo(
+                FILTER_BUILDER.allOf(
+                    inputFilter,
+                    FILTER_BUILDER.anyOf(
+                        FILTER_BUILDER
+                            .attribute(Core.METACARD_OWNER)
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_INDIVIDUALS)
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_INDIVIDUALS_READ)
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_ADMINISTRATORS)
+                            .is()
+                            .equalTo()
+                            .text(EMAIL_BOB),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_A),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS_READ)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_A),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_B),
+                        FILTER_BUILDER
+                            .attribute(Security.ACCESS_GROUPS_READ)
+                            .is()
+                            .equalTo()
+                            .text(ROLE_B))))));
+  }
+
+  private void verifyPluginDoesNotAlterTheFilter(Filter inputFilter)
+      throws PluginExecutionException, StopProcessingException {
+    QueryRequest request = plugin.process(new QueryRequestImpl(new QueryImpl(inputFilter)));
+    assertThat(asGeotoolsComparableFilter(request.getQuery()), is(equalTo(inputFilter)));
+  }
+
+  private static Filter asGeotoolsComparableFilter(Query query) {
+    return (Filter) query.accept(COPYING_VISITOR, null);
+  }
+
+  private static class AccessControlPreQueryPluginUnderTest extends AccessControlPreQueryPlugin {
+    private final List<String> subjectRoles;
+
+    private AccessControlPreQueryPluginUnderTest(
+        FilterBuilder filterBuilder, String... subjectRoles) {
+      super(
+          filterBuilder,
+          null /* Only method interacting with this ref gets overridden */,
+          config(""));
+      this.subjectRoles = Arrays.asList(subjectRoles);
+      this.init();
+    }
+
+    private AccessControlPreQueryPluginUnderTest(
```

Comment:
❗️ The amount of setup work required to test the `AccessControlPreQueryPlugin` feels like a design smell. Some of the complexity seems to be stemming from the fact that the plugin is also doing the work of aggregating the access controlled tags from the system, forcing us to also mock `ServiceReferences`. What do you think about having another class (could name it `AccessControlTags`) do that heavy lifting for us? Then you can just mock one its methods (maybe `Set<String> getTags()`) and hardcode your own tags for your tests.

Reply:
That's actually a really good idea. I'll look into it.

Reply:
Reworked. Will also bring over some unit tests.